rpcs3/Emu/RSX/Overlays/overlay_osk.cpp

```diff
@@ -641,12 +649,6 @@ namespace rsx
 		void osk_dialog::on_backspace(const std::u32string&)
 		{
 			m_preview.erase();
-
-			if (m_preview.text.empty())
```

Comment:
It was implemented intentionally to restore some kind of hint text if completely empty. Not text to be returned, but like how textboxes and search bars work in modern UIs. If I remove all the text from this github comment text area, hint text saying 'Leave a comment' will appear every time.

Reply:
No. this was actually my addition that caused the regression of the attached issue.
The original code only showed the placeholder initially.
If you look at the set_text overrides in "overlay_edit_text" then you will see that It now properly handles the exact behavior you are describing.
Since I moved the logic to the text edit, we can now also use it in other future applications natively.